service/api/service.go

```diff
@@ -79,13 +79,15 @@ func (s *serviceImpl) ApiV1WorkflowStartPost(ctx context.Context, req iwfidl.Wor
 			workflowConfig = startOptions.GetWorkflowConfigOverride()
 		}
 		if startOptions.GetUseMemoForDataAttributes() {
-			workflowOptions.Memo = map[string]interface{}{
-				service.WorkerUrlMemoKey: iwfidl.EncodedObject{
-					Data: iwfidl.PtrString(req.IwfWorkerUrl), // this is hack to ensure all memos are with the same type
-				},
-			}
 			useMemo = true
 		}
+
+		// the workerUrl will always be stored in the memo regardless of `startOptions.GetUseMemoForDataAttributes()`
```

Comment:
I think we need to move it one more layer out. Even there is not startOptions, we still need to store the workerUrl into memo